scripts/add-icon.js

```diff
@@ -0,0 +1,71 @@
+import fs from 'node:fs/promises';
+import inquirer from 'inquirer';
+import chalkPipe from 'chalk-pipe';
+import logSymbols from 'log-symbols';
+import getRelativeLuminance from 'get-relative-luminance';
+import { getIconsDataString, getIconDataPath, titleToSlug } from './utils.js';
+
+const hexPattern = /^#?[a-fA-F0-9]{6}$/;
+const iconsData = JSON.parse(await getIconsDataString());
+
+const prompts = [
+  {
+    type: 'input',
+    name: 'title',
+    validate: (text) => {
+      if (!text) return 'This field is required';
+      if (iconsData.icons.find((x) => x.title === text))
+        return 'This icon title already exist';
+      return true;
+    },
+    message: 'Title',
+  },
+  {
+    type: 'input',
+    name: 'hex',
+    message: 'Hex',
+    validate: (text) =>
+      hexPattern.test(text) ? true : 'This should be a 6-digit hex code',
+    transformer: (text) => {
+      const color = text.startsWith('#') ? text : `#${text}`;
+      const luminance = hexPattern.test(text)
+        ? getRelativeLuminance.default(color)
+        : -1;
+      if (luminance === -1) return text;
+      return chalkPipe(`bg${color}.${luminance < 0.4 ? 'white' : 'black'}`)(
+        text,
+      );
+    },
+  },
+  {
+    type: 'input',
+    name: 'source',
+    message: 'Source',
+    validate: (text) =>
+      Boolean(text.startsWith('https://') || text.startsWith('http://'))
+        ? true
+        : 'This should be a URL',
+  },
+];
+
+const icon = await inquirer.prompt(prompts);
```

Comment:
After this, the icon will be added without a confirmation which is a destructive change. What about a final `confirm` question?

```suggestion
  },
  {
    type: 'confirm',
    message: (answers) => {
      return `The filename of the icon would be`
           + ` ${chalkPipe('blue')(titleToSlug(answers.title) + '.svg')}\n`
           + 'Add you sure that you want to add it to the data file?';
    },
    name: 'confirmation',
  },
];

const icon = await inquirer.prompt(prompts);
if (!icon.confirmation) {
  console.log(logSymbols.error, 'Icon not added');
  process.exit(1);
}
```

And we can remove the part `your icon filename should be: ...` of the final messsage.

Reply:
Sounds good. The final confirm question is a good idea.

> And we can remove the part your icon filename should be: ... of the final message.

No problem.